### PR TITLE
fix: set the exit code for json output

### PR DESCRIPTION
**Pull request by barrett-schonefeld**
Purpose:
- Want the exit code set properly for json output.

Changes:
- Move the logic to set the exit code so that it applies to json output as well.
- If the warnings limit exceeded, add this as an error to the results so that it may be naturally included in the json output.

Tests:
- repair warnings limit tests and exit code tests

### CODE REVIEW

File: src/cli-validator/runValidator.js [hunk 1]
```diff
@@ -288,6 +288,33 @@ const processInput = async function(program) {
...
+        if (!results.errors) {
+          results.errors = {};
+        }
+        results.errors['warnings-limit'] = [
+          {
+            path: [],
+            message: `Number of warnings (${numWarnings}) exceeds warnings limit (${
+              limitsObject.warnings
+            }).`,
+            rule: 'warnings_limit'
```
Comment (dpopp07) [inline src/cli-validator/runValidator.js:312]: I may be overthinking this but I wonder if this isn't the most clear. We don't actually have a rule anywhere called "warnings_limit" so I don't want to mislead users into thinking that they should be setting that property somewhere. I suppose that if they see this message, they already know how to configure it, and I don't have a better option. Perhaps this error doesn't need a "rule" though.

**Review by dpopp07 — APPROVED** [review]
I left one comment, but looks good! I like the idea of including the warning limit message in the list of errors instead of a console log 👍

File: src/cli-validator/runValidator.js (hunk 1 above)
Comment (barrett-schonefeld) [reply src/cli-validator/runValidator.js:312]: I think you're right. This would be better suited as "builtin"

**Pull request merged**